Panic in checkAndRemoveUnderscoresIntegers on trailing plus

**Describe the bug**
Given `0=+`, `go-toml` panics:
```
$ echo -n "0=+" | gotoml-test-decoder
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/pelletier/go-toml/v2.checkAndRemoveUnderscoresIntegers({0xc000134002, 0x6beca3a84a68, 0xe0})
        /x/src/github.com/pelletier/go-toml/decode.go:403 +0x2e8
github.com/pelletier/go-toml/v2.parseIntDec({0xc000134002, 0x1, 0x1fe})
        /x/src/github.com/pelletier/go-toml/decode.go:374 +0x2c
github.com/pelletier/go-toml/v2.parseInteger({0xc000134002, 0x0, 0xc000136070})
        /x/src/github.com/pelletier/go-toml/decode.go:24 +0x74
github.com/pelletier/go-toml/v2.(*decoder).unmarshalInteger(0xc000138000, 0xc00013a070, {0x4e02e0, 0xc00010c300, 0x10})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:872 +0x67
github.com/pelletier/go-toml/v2.(*decoder).handleValue(0x4ddd01, 0xc00013a070, {0x4e02e0, 0xc00010c300, 0x10})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:636 +0x1ac
github.com/pelletier/go-toml/v2.(*decoder).handleKeyValueInner(0x4da540, {0x1, 0x0}, 0x4ddd40, {0x4e02e0, 0xc00010c300, 0x6bec7cdf5208})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:986 +0x8c
github.com/pelletier/go-toml/v2.(*decoder).handleKeyValuePart(0xc000138000, {0x1, 0xc00013a038}, 0x5c9be0, {0x4e1d80, 0xc000124018, 0xffffffffffffffff})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:1030 +0x54b
github.com/pelletier/go-toml/v2.(*decoder).handleKeyValueInner(0x1, {0x1, 0xc00013a038}, 0xc000108c28, {0x4e1d80, 0xc000124018, 0xc00013a000})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:982 +0x71
github.com/pelletier/go-toml/v2.(*decoder).handleKeyValue(0xc000138000, 0xc00013a000, {0x4e1d80, 0xc000124018, 0x200})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:968 +0xa5
github.com/pelletier/go-toml/v2.(*decoder).handleRootExpression(0xc000138000, 0xc00013a000, {0x4e1d80, 0xc000124018, 0x0})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:257 +0xa9
github.com/pelletier/go-toml/v2.(*decoder).fromParser(0xc000138000, {0x4e1d80, 0xc000124018, 0x100010000488a26})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:223 +0xa9
github.com/pelletier/go-toml/v2.(*decoder).FromParser(0xc000138000, {0x4da800, 0xc000124018})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:208 +0x1e7
github.com/pelletier/go-toml/v2.(*Decoder).Decode(0xc000108ec8, {0x4da800, 0xc000124018})
        /x/src/github.com/pelletier/go-toml/unmarshaler.go:112 +0x1d2
github.com/pelletier/go-toml/v2/testsuite.DecodeStdin()
        /x/src/github.com/pelletier/go-toml/testsuite/testsuite.go:39 +0x7c
main.main()
        /x/src/github.com/pelletier/go-toml/cmd/gotoml-test-decoder/main.go:20 +0xb9
```

**To Reproduce**
https://go.dev/play/p/ocOmu3sNFzz
```go
toml.Unmarshal([]byte("0=+"), &v)
```

**Expected behavior**
Not a panic.  Expected error for invalid input.

**Versions**
 - go-toml: d58eb50ebfb929b84ef2af730b2f8503aa4a25fb
 - go version go1.17.5 linux/amd64

**Additional context**
Found while fuzzing (but not differential fuzzing this time).

Related?

Given `[]byte("0=+A")`:

```
$ gotip test -fuzz=Num
fuzz: elapsed: 0s, gathering baseline coverage: 0/64 completed                                                                                                                   [7/474]
failure while testing seed corpus entry: FuzzUnmarshalNumbers/d87d8747edeae4dab064f6dd5827cc6dc261cc9c474a8b5bedfbf9589f823682
fuzz: elapsed: 0s, gathering baseline coverage: 14/64 completed
--- FAIL: FuzzUnmarshalNumbers (0.04s)
    --- FAIL: FuzzUnmarshalNumbers (0.00s)
        testing.go:1350: panic: runtime error: index out of range [1] with length 1
            goroutine 52 [running]:
            runtime/debug.Stack()
                /x/sdk/gotip/src/runtime/debug/stack.go:24 +0x90
            testing.tRunner.func1()
                /x/sdk/gotip/src/testing/testing.go:1350 +0x1f2
            panic({0x735160, 0xc0003cc0d8})
                /x/sdk/gotip/src/runtime/panic.go:838 +0x207
            github.com/pelletier/go-toml/v2.checkAndRemoveUnderscoresIntegers({0xc00039876a, 0x1, 0x6})
                /x/src/github.com/pelletier/go-toml/decode.go:403 +0x5af
            github.com/pelletier/go-toml/v2.parseIntDec({0xc00039876a, 0x1, 0x6})
                /x/src/github.com/pelletier/go-toml/decode.go:374 +0x45
            github.com/pelletier/go-toml/v2.parseInteger({0xc00039876a, 0x1, 0x6})
                /x/src/github.com/pelletier/go-toml/decode.go:24 +0xa5
            github.com/pelletier/go-toml/v2.(*decoder).unmarshalInteger(0xc00038a300?, 0xc0003e83f0?, {0x701f20?, 0xc00038c780?, 0x40ea65?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:872 +0x7f
            github.com/pelletier/go-toml/v2.(*decoder).handleValue(0xc00038c780?, 0xc0003e83f0, {0x701f20?, 0xc00038c780?, 0x462a67?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:636 +0x34f
            github.com/pelletier/go-toml/v2.(*decoder).handleKeyValueInner(0x6f1100?, {0x1, 0x0}, 0x6f9da0?, {0x701f20?, 0xc00038c780?, 0x1?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:986 +0x199
            github.com/pelletier/go-toml/v2.(*decoder).handleKeyValuePart(0xc00038a300, {0x1, 0xc0003e83b8}, 0x6?, {0x707460?, 0xc000396c60?, 0x6a2ac5?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:1030 +0xa8b
            github.com/pelletier/go-toml/v2.(*decoder).handleKeyValuePart(0xc00038a300, {0x1, 0xc0003e83b8}, 0x5c15b7?, {0x701f20?, 0xc00038c760?, 0x2?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:1078 +0x206
            github.com/pelletier/go-toml/v2.(*decoder).handleKeyValueInner(0xc0003e8380?, {0x1, 0xc0003e83b8}, 0x2?, {0x701f20?, 0xc00038c760?, 0xc00038a318?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:982 +0x16f
            github.com/pelletier/go-toml/v2.(*decoder).handleKeyValue(0xc00038a300, 0xc0003e8380, {0x701f20?, 0xc00038c760?, 0xc00039876b?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:968 +0x125
            github.com/pelletier/go-toml/v2.(*decoder).handleRootExpression(0xc00038a300, 0xc0003e8380, {0x701f20?, 0xc00038c760?, 0x701f20?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:257 +0x276
            github.com/pelletier/go-toml/v2.(*decoder).fromParser(0xc00038a300, {0x701f20?, 0xc00038c760?, 0x707460?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:223 +0xf3
            github.com/pelletier/go-toml/v2.(*decoder).FromParser(0xc00038a300, {0x6f1100?, 0xc00038c760?})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:208 +0x4e7
            github.com/pelletier/go-toml/v2.Unmarshal({0xc000398768, 0x4, 0x8}, {0x6f1100, 0xc00038c760})
                /x/src/github.com/pelletier/go-toml/unmarshaler.go:28 +0x137
            github.com/pelletier/go-toml/v2.FuzzUnmarshalNumbers.func1(0xc000381040, {0xc000398768, 0x4, 0x8})
                /x/src/github.com/pelletier/go-toml/fuzz_test.go:103 +0x1bd
            reflect.Value.call({0x6fdb60?, 0xc000099070?, 0x13?}, {0x74924c, 0x4}, {0xc000396c30, 0x2, 0x2?})
                /x/sdk/gotip/src/reflect/value.go:556 +0x845
            reflect.Value.Call({0x6fdb60?, 0xc000099070?, 0x513580?}, {0xc000396c30, 0x2, 0x2})
                /x/sdk/gotip/src/reflect/value.go:339 +0xbf
            testing.(*F).Fuzz.func1.1(0x0?)
                /x/sdk/gotip/src/testing/fuzz.go:332 +0x20b
            testing.tRunner(0xc000381040, 0xc0003a87e0)
                /x/sdk/gotip/src/testing/testing.go:1440 +0x102
            created by testing.(*F).Fuzz.func1
                /x/sdk/gotip/src/testing/fuzz.go:321 +0x5b8

FAIL
exit status 1
FAIL    github.com/pelletier/go-toml/v2 0.112s
```
```
$ cat testdata/fuzz/FuzzUnmarshalNumbers/d87d8747edeae4dab064f6dd5827cc6dc261cc9c474a8b5bedfbf9589f823682
go test fuzz v1
[]byte("0=+A")
```

--- a/decode.go
+++ b/decode.go
@@ -400,6 +400,10 @@ func checkAndRemoveUnderscoresIntegers(b []byte) ([]byte, error) {
 		start++
 	}
 
+	if len(b) == start {
+		return b, nil
+	}
+
 	if b[start] == '_' {
 		return nil, newDecodeError(b[start:start+1], "number cannot start with underscore")
 	}

--- a/unmarshaler_test.go
+++ b/unmarshaler_test.go
@@ -2326,6 +2326,18 @@ func TestIssue710(t *testing.T) {
 	require.Equal(t, map[string]toml.LocalTime{"0": {Nanosecond: 111111111, Precision: 9}}, v2)
 }
 
+func TestIssue715(t *testing.T) {
+	var v interface{}
+	err := toml.Unmarshal([]byte("0=+"), &v)
+	require.Error(t, err)
+
+	err = toml.Unmarshal([]byte("0=-"), &v)
+	require.Error(t, err)
+
+	err = toml.Unmarshal([]byte("0=+A"), &v)
+	require.Error(t, err)
+}
+
 func TestUnmarshalDecodeErrors(t *testing.T) {
 	examples := []struct {
 		desc string

